cmd/minikube/cmd/config/util.go

```diff
@@ -138,7 +139,17 @@ func EnableOrDisableAddon(name string, val string) error {
 
 func enableOrDisableAddonInternal(addon *assets.Addon, cmd bootstrapper.CommandRunner, data interface{}, enable bool) error {
 	var err error
+	// check addon status before enable/disable
+	ok, err := addon.IsEnabled()
+	if err != nil {
+		exit.WithError("Failed to get the addon status ", err)
+	}
+
 	if enable {
+		if ok {
```

Comment:
I think this logic can be moved to the respective command files.

Reply:
Address done. Thanks